### DISCUSSION TITLE
Kindly help me understand why this client goes offline randomly

### DISCUSSION THREAD

**Discussion by acrolink** [strongswan/strongswan#667] [Q&A]
One client connects successfully but after half and hours - hour goes offline. Here what I see in the logs:

```
Thu, 2021-10-14, 03:01:41 14[NET] <office-conn|18> sending packet: from my.server.ip.addr[4500] to clients.public.ip.addr[1124] (76 bytes)
Thu, 2021-10-14, 03:01:50 11[NET] <office-conn|18> received packet: from clients.public.ip.addr[1124] to my.server.ip.addr[4500] (76 bytes)
Thu, 2021-10-14, 03:01:50 11[ENC] <office-conn|18> parsed INFORMATIONAL request 116 [ ]
Thu, 2021-10-14, 03:01:50 11[ENC] <office-conn|18> generating INFORMATIONAL response 116 [ ]
Thu, 2021-10-14, 03:01:50 11[NET] <office-conn|18> sending packet: from my.server.ip.addr[4500] to clients.public.ip.addr[1124] (76 bytes)
Thu, 2021-10-14, 03:01:55 08[KNL] received a XFRM_MSG_EXPIRE
Thu, 2021-10-14, 03:01:55 08[KNL] creating delete job for CHILD_SA ESP/0xc68a6e93/my.server.ip.addr
Thu, 2021-10-14, 03:01:55 08[JOB] CHILD_SA ESP/0xc68a6e93/my.server.ip.addr not found for delete
Thu, 2021-10-14, 03:01:55 05[IKE] <conn-10-182-10-12|15> giving up after 5 retransmits
Thu, 2021-10-14, 03:01:55 05[IKE] <conn-10-182-10-12|15> IKE_SA conn-10-182-10-12[15] state change: ESTABLISHED => DESTROYING
Thu, 2021-10-14, 03:01:55 05[CHD] <conn-10-182-10-12|15> CHILD_SA net{57} state change: CREATED => DESTROYING
Thu, 2021-10-14, 03:01:55 05[KNL] <conn-10-182-10-12|15> deleting SAD entry with SPI c68a6e93
Thu, 2021-10-14, 03:01:55 05[CHD] <conn-10-182-10-12|15> CHILD_SA net{43} state change: REKEYING => DESTROYING
Thu, 2021-10-14, 03:01:55 05[KNL] <conn-10-182-10-12|15> deleting policy 10.10.10.0/29 === 10.182.10.12/32 out
Thu, 2021-10-14, 03:01:55 05[KNL] <conn-10-182-10-12|15> deleting policy 10.182.10.12/32 === 10.10.10.0/29 in
Thu, 2021-10-14, 03:01:55 05[KNL] <conn-10-182-10-12|15> deleting policy 10.182.10.12/32 === 10.10.10.0/29 fwd
Thu, 2021-10-14, 03:01:55 05[KNL] <conn-10-182-10-12|15> deleting policy 10.182.10.0/24 === 10.182.10.12/32 out
Thu, 2021-10-14, 03:01:55 05[KNL] <conn-10-182-10-12|15> deleting policy 10.182.10.12/32 === 10.182.10.0/24 in
Thu, 2021-10-14, 03:01:55 05[KNL] <conn-10-182-10-12|15> deleting policy 10.182.10.12/32 === 10.182.10.0/24 fwd
Thu, 2021-10-14, 03:01:55 05[KNL] <conn-10-182-10-12|15> deleting SAD entry with SPI c0fc63bc
Thu, 2021-10-14, 03:01:55 05[KNL] <conn-10-182-10-12|15> deleted SAD entry with SPI c0fc63bc
Thu, 2021-10-14, 03:01:55 05[KNL] <conn-10-182-10-12|15> deleting SAD entry with SPI 3f522fa6
Thu, 2021-10-14, 03:01:55 05[KNL] <conn-10-182-10-12|15> deleted SAD entry with SPI 3f522fa6
Thu, 2021-10-14, 03:01:55 05[CFG] <conn-10-182-10-12|15> lease 10.182.10.12 by 'eap_10_182_10_12' went offline
```
and:

```
Oct 14 03:01:50 localhost charon-systemd[675]: received packet: from clients.public.ip.addr[1124] to my.server.ip.addr[4500] (76 bytes)
Oct 14 03:01:50 localhost charon-systemd[675]: parsed INFORMATIONAL request 116 [ ]
Oct 14 03:01:50 localhost charon-systemd[675]: generating INFORMATIONAL response 116 [ ]
Oct 14 03:01:50 localhost charon-systemd[675]: sending packet: from my.server.ip.addr[4500] to clients.public.ip.addr[1124] (76 bytes)
Oct 14 03:01:55 localhost charon-systemd[675]: creating delete job for CHILD_SA ESP/0xc68a6e93/my.server.ip.addr
Oct 14 03:01:55 localhost charon-systemd[675]: CHILD_SA ESP/0xc68a6e93/my.server.ip.addr not found for delete
Oct 14 03:01:55 localhost charon-systemd[675]: giving up after 5 retransmits
Oct 14 03:01:55 localhost charon-systemd[675]: lease 10.182.10.12 by 'eap_10_182_10_12' went offline
```
Worth to mention that clients `conn-10-182-10-12` and `office-conn` are behind the same remote public ip address.

What might have caused the disconnection of `conn-10-182-10-12`? Thank you.

**Comment by tobiasbrunner** [comment]
Difficult to tell. The complete log might help (also from the other end).

**Comment by acrolink** [comment]
Is there a reason that for some devices [android] this `ESP:AES_CBC-128/HMAC_SHA1_96` can be more stable than this: `ESP:AES_GCM_16-128/MODP_3072`?

I have made the above switch for a device and it seems more stable with no connection drops now.

**Reply by tobiasbrunner** [reply]
Please read [this](https://wiki.strongswan.org/projects/strongswan/wiki/ExpiryRekey#IKEv2).

**Comment by acrolink** [comment]
Thanks @tobiasbrunner , your help is highly appreciated. I am trying to improve the connectivity for my clients and reduce connection drops.

What I have done now is:

1. Reduced rekeying time for both `IKE` and `Child SA`.
2. Enabled mobike.
3. A server connected to the same VPN pings other clients in the VPN every minute.

My question, is the above reasonable or is there something that is redundant?

Thank you.

**Reply by tobiasbrunner** [reply]
> Reduced rekeying time for both `IKE` and `Child SA`.

Not sure what that has to do with connection drops or connectivity (depending on what you configured, it might even have a negative effect  e.g. if a mobile client isn't reachable when the server wants to do a rekeying).

> Enabled mobike.

Definitely a good idea if you have mobile clients.

>  A server connected to the same VPN pings other clients in the VPN every minute.

To what end?

**Reply by acrolink** [reply]
> > A server connected to the same VPN pings other clients in the VPN every minute.
> 
> To what end?

Say I have a VPN client with connection drops (A). Another VPN client (B) is set to ping (A) every minute to keep it alive. Not sure if that makes sense.

What I suspect in the meanwhile is that reducing `MTU` can help.

One question,

Is setting something like this using `iptables` on the VPN server, makes it not necessary to lower and set the `MTU` on the Android client to `1280`? Or should both server and client be configured to use the value `1280`?

```
-A FORWARD -p tcp -m policy --dir in --pol ipsec -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1281:1536  -j TCPMSS --set-mss 1280
-A FORWARD -p tcp -m policy --dir out --pol ipsec -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1281:1536  -j TCPMSS --set-mss 1280
```

**Reply by tobiasbrunner** [reply]
> Say I have a VPN client with connection drops (A). Another VPN client (B) is set to ping (A) every minute to keep it alive. Not sure if that makes sense.

If client A is not reachable, how does a constant ping help? Pings are useful if there is an inactivity timeout (SAs get closed if there is no traffic), or if SAs on B are triggered by traffic (i.e. via trap policies). If A is a mobile device, a constant ping might keep from going to sleep, but is that what you want (might drain A's battery more quickly)?


> Is setting something like this using `iptables` on the VPN server, makes it not necessary to lower and set the `MTU` on the Android client to `1280`? Or should both server and client be configured to use the value `1280`?

For TCP traffic it's enough as the rules modify the TCP headers that pass through. Lowering the MTU on the client can help for UDP traffic, but it would require similar adjustments on the server for response packets.

**Reply by acrolink** [reply]
> If A is a mobile device, a constant ping might keep from going to sleep, but is that what you want (might drain A's battery more quickly)?

That is exactly what I want. I want to keep the connection alive in the Android client (A). That's the point. It has constant power supply.

I will give it a try to see if making A ping B constantly, if that would keep A online on the VPN.

I guess, A needs to implement something like this:

```
    auto=route
    closeaction=restart
```
But I am not sure if the Android client currently supports that.

Thanks again for the help.

**Reply by tobiasbrunner** [reply]
> But I am not sure if the Android client currently supports that.

If you are referring to the Android app, it does that anyway. Once a connection is started, it will be kept up until the user manually disconnects it.
